### ISSUE TITLE
Masking for concatenating multiple inputs' embedding

### ISSUE THREAD

**Issue by snakeztc** [fchollet/keras#2101]
I am working on a model that takes 2 time series inputs X1 and X2 with variable length. X1 and X2 will first be projected to their own embedding using timeDistributedDense layer and the embedding are concatenated together and feed into an LSTM. My **issue is that the masking is lost when I concatenate the two embedding**. How can I force a mask to the concatenated output? My code is as follows: Thank you very much! I just begin to use Keras and its awesome.

```
x1_embed = 10
x2_embed = 20
embed_size = x1_embed + x2_embed 

graph.add_input(name='x1', input_shape=(None, x1_size))
graph.add_input(name='x2', input_shape=(None, x2_size))

# add masking
graph.add_node(Masking(mask_value=0.0, input_shape=(None, x1_size)), name="x1_mask", input="x1")
graph.add_node(Masking(mask_value=0.0, input_shape=(None, x2_size)), name="x2_mask", input="x2")

# add embedding
graph.add_node(TimeDistributedDense(sys_embed, input_dim=sys_size), name="x1_embed", input="x1_mask")
graph.add_node(TimeDistributedDense(usr_embed, input_dim=usr_size), name="x2_embed", input="x2_mask")

# I thought this will enforce the mask, but it DOES NOT
graph.add_node(Masking(mask_value=0.0, input_shape=(None, embed_size)),
               inputs=["x1_embed", "x2_embed"], name="x1x2_embed", merge_mode='concat')

shared_model = containers.Sequential()
shared_model.add(LSTM(256, input_dim=embed_size, return_sequences=False))
shared_model.add(Dropout(0.3))

graph.add_node(shared_model, name="recurrent_layers", input="turn_embed")

graph.add_node(Dense(100, activation='tanh'), name="output", input="recurrent_layers")

graph.compile(optimizer=opt, loss={"output": "mse"})
```

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by EderSantana** [comment]
I don't think Keras supports heterogenous masking 

> graph.add_node(Masking(mask_value=0.0, input_shape=(None, embed_size)),
>                inputs=["x1_embed", "x2_embed"], name="x1x2_embed", merge_mode='concat')

This will probably just consider the masking of the largest sequence

**Comment by snakeztc** [comment]
@EderSantana Thank you very much for the reply. Do you know any work around to this problem?

**Comment by fchollet** [comment]
There will be support for Merge masking in the near future. Support won't
be added to Keras 1.0 from the get-go, but will be easy to add in the first
few weeks after the release (it's on the TODO).

**Comment by voletiv** [comment]
Could you please let us know the progress of Mask support for Merge layers? When is it likely to be implemented?
